contracts/Overtime/SportsAMMV2.sol

```diff
@@ -445,61 +508,91 @@ contract SportsAMMV2 is Initializable, ProxyOwned, ProxyPausable, ProxyReentranc
+            collateralPrice = ICollateralUtility(ICollateralUtility(address(multiCollateralOnOffRamp)).priceFeed())
+                .rateForCurrency(ISportsAMMV2LiquidityPool(lqPool).collateralKey());
         } else {
+            uint collateralQuote = multiCollateralOnOffRamp.getMinimumNeeded(_collateral, _buyInAmount);
             IERC20(_collateral).safeTransferFrom(_fromAddress, address(this), collateralQuote);
             IERC20(_collateral).approve(address(multiCollateralOnOffRamp), collateralQuote);
             exactReceived = multiCollateralOnOffRamp.onramp(_collateral, collateralQuote);
         }
 
-        require(exactReceived >= _buyInAmount, "Not enough default payment token received");
+        require(exactReceived >= _buyInAmount, "Insuff payment");
 
         //send the surplus to SB
         if (exactReceived > _buyInAmount) {
-            defaultCollateral.safeTransfer(safeBox, exactReceived - _buyInAmount);
+            if (lqPool != address(0)) {
+                // Note: if needed add the logic to convert the surplus to default collateral and send to safeBox
+                IERC20(_collateral).safeTransfer(safeBox, exactReceived - _buyInAmount);
+            } else {
+                defaultCollateral.safeTransfer(safeBox, exactReceived - _buyInAmount);
+            }
         }
     }
 
     function _trade(ISportsAMMV2.TradeData[] memory _tradeData, TradeDataInternal memory _tradeDataInternal) internal {
         uint totalQuote = (ONE * _tradeDataInternal._buyInAmount) / _tradeDataInternal._expectedPayout;
         uint payout = _tradeDataInternal._expectedPayout;
-        //TODO: include this in the tradeQuote method for live trading
         uint fees = (safeBoxFee * _tradeDataInternal._buyInAmount) / ONE;
-        uint[] memory amountsToBuy = new uint[](_tradeData.length);
 
         if (!_tradeDataInternal._isLive) {
-            (totalQuote, payout, fees, amountsToBuy, ) = _tradeQuote(_tradeData, _tradeDataInternal._buyInAmount, false);
-        } else {
-            amountsToBuy[0] = payout;
+            (totalQuote, payout, fees, , ) = _tradeQuote(_tradeData, _tradeDataInternal._buyInAmount, false, 0);
         }
 
         uint payoutWithFees = payout + fees;
-
-        riskManager.checkAndUpdateRisks(_tradeData, _tradeDataInternal._buyInAmount);
-        riskManager.checkLimits(
-            _tradeDataInternal._buyInAmount,
-            totalQuote,
-            payout,
-            _tradeDataInternal._expectedPayout,
-            _tradeDataInternal._additionalSlippage
-        );
-
-        if (_tradeDataInternal._sendDefaultCollateral) {
-            defaultCollateral.safeTransferFrom(
-                _tradeDataInternal._requester,
-                address(this),
-                _tradeDataInternal._buyInAmount
+        uint transformDecimal = (18 - defaultCollateralDecimals);
+        if (_tradeDataInternal._collateralPool == address(0)) {
```

Comment:
I prefer if this doesnt have as much repetative code, seems like the same thing is done more or less in both If and ELSE branches

Reply:
Modified the code, added another function :
`_checkRisksAndUpdateStakingVolume`